build_zng.rs

```diff
@@ -1,58 +1,691 @@
-use std::env;
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    env, fs,
+    hash::{Hash as _, Hasher as _},
+    io::Write as _,
+    path::{Path, PathBuf},
+};
+
+fn append(cfg: &mut cc::Build, root: Option<&str>, files: impl IntoIterator<Item = &'static str>) {
```

Comment:
Recommends to use trait object to speedup compilation:

```suggestion
fn append(cfg: &mut cc::Build, root: Option<&str>, files: &dyn mut Iterator<Item = &'static str>) {
```

Reply:
Any speedup would not be noticeable, I don't see the point.